client/src/app/admin/panel/settings/admin-panel-advanced-settings.js

```diff
@@ -156,24 +162,47 @@ class AdminPanelAdvancedSettings extends React.Component {
             <Form className="admin-panel-advanced-settings__api-keys-modal" onSubmit={this.addAPIKey.bind(this)}>
                 <Header title={i18n('ADD_API_KEY')} description={i18n('ADD_API_KEY_DESCRIPTION')}/>
                 <FormField name="name" label={i18n('NAME_OF_KEY')} validation="DEFAULT" required fieldProps={{size: 'large'}}/>
-                <SubmitButton type="secondary">{i18n('SUBMIT')}</SubmitButton>
+                <div className="admin-panel-advanced-settings__api-keys__permissions">
+                    <FormField className = "admin-panel-advanced-settings__api-keys__permissions__item" name="createTicketPermission" label={i18n('TICKET_CREATION_PERMISSION')}   field='checkbox'/>
+                    <FormField name="ticketNumberPermission" label={i18n('TICKET_NUMBER_RETURN_PERMISSION')}  field='checkbox'/>
+                </div>
+                <div className="admin-panel-advanced-settings__api-keys__permissions" >
+                    <FormField className = "admin-panel-advanced-settings__api-keys__permissions__item" name="commentTicketPermission" label={i18n('TICKET_COMMENT_PERMISSION')}  field='checkbox'/>
+                    <FormField name="userPermission" label={i18n('USER_CREATION_PERMISSION')}   field='checkbox'/>
+                </div>
+                <SubmitButton className="admin-panel-advanced-settings__api-keys-modal__submit-button" type="secondary">{i18n('SUBMIT')}</SubmitButton>
             </Form>
         );
     }
 
-    addAPIKey({name}) {
+    addAPIKey({name,userPermission,createTicketPermission,commentTicketPermission,ticketNumberPermission}) {
         ModalContainer.closeModal();
+        this.setState({
+            error: ''
+        })
         API.call({
             path: '/system/add-api-key',
-            data: {name, type: 'REGISTRATION'}
-        }).then(this.getAllKeys.bind(this));
+            data: {
+                name,
+                canCreateUsers: userPermission*1,
+                canCreateTickets: createTicketPermission*1,
+                canCommentTickets: commentTicketPermission*1,
+                shouldReturnTicketNumber: ticketNumberPermission*1
+            }
+        }).then(this.getAllKeys.bind(this)).catch((result) =>{
+            this.setState({
+                error: result.message
+            })
+        });
     }
 
     getAllKeys() {
         API.call({
             path: '/system/get-api-keys',
             data: {}
-        }).then(this.onRetrieveSuccess.bind(this));
+        }).then(this.onRetrieveSuccess.bind(this)).catch((result)=> {
+            console.log(result.message)
```

Comment:
We probably need a frontend part for when this fails, or, at least, wrap this checking for the `showLogs` variable.